### DISCUSSION TITLE
Cannot find module 'next-auth/providers/email' or its corresponding type declarations.

### DISCUSSION THREAD

**Discussion by jrdurant** [nextauthjs/next-auth#3262] [Help]
### Question 💬

Hopefully not silly, but I am attempting to use some providers in my typescript project. Here is my [...nextAuth].ts file, straight from the example https://next-auth.js.org/.

I am getting the error from the title TS2307: Cannot find module 'next-auth/providers/google' or its corresponding type declarations. Am I doing something wrong? 

I am using the "next-auth": "4.0.0-beta.2"

`import NextAuth from 'next-auth';
import GoogleProvider from 'next-auth/providers/google';
import EmailProvider from 'next-auth/providers/email';

export default NextAuth({
  providers: [
    GoogleProvider({
      clientId: process.env.GOOGLE_ID,
      clientSecret: process.env.GOOGLE_SECRET,
    }),
    // Passwordless / email sign in
    EmailProvider({
      server: process.env.MAIL_SERVER,
      from: 'NextAuth.js <no-reply@example.com>',
    }),
  ],
});
`

### How to reproduce ☕️

Follow the instructions from here: https://next-auth.js.org/, but in a typescript file instead of .js. 

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by Izhaki** [comment]
I have just renamed `[...nextauth].js` > `[...nextauth].ts` with `"next-auth": "4.0.0-beta.7"` and no problems (I get different build errors with `4.0.0-beta.2`.

I have a feeling updating to `4.0.0-beta.7` will solve this issue. If not then I guess something is wrong with your build settings, so please share how exactly do you build your project (or which sample project you use)

**Reply by balazsorban44** [reply]
The repo is open (source). Feel free to check yourself. Updating to beta.7 is a good idea anyway.

**Reply by jrdurant** [reply]
I started with the beta.7. I think my issue was including the ^ in front of the version. Deleting my node_modules and trying again helped.

**Reply by balazsorban44** [reply]
Sorry, there is a note about that in the migration: https://next-auth.js.org/getting-started/upgrade-v4#verify-the-correct-version
